Gems/Atom/Feature/Common/Code/Source/Decals/DecalTextureArrayFeatureProcessor.h

```diff
@@ -143,8 +145,10 @@ namespace AZ
             void CullDecals(const RPI::ViewPtr& view);
             // Get or create a buffer that will be used for visibility when doing CPU culling.
             GpuBufferHandler& GetOrCreateVisibleBuffer();
+            void UpdateBounds(const DecalHandle handle);
 
-            IndexedDataVector<DecalData> m_decalData;
+            MultiIndexedDataVector<DecalData, AZ::Aabb> m_decalData;
```

Comment:
out of curiosity, what is this container?

Reply:
It is implemented here - https://github.com/o3de/o3de/blob/development/Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Utils/MultiIndexedDataVector.h. It is an extension of https://github.com/o3de/o3de/blob/development/Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Utils/IndexedDataVector.h where the data is stored in a gpu friendly way.